examples/nextjs-14-permit/src/app/api/stats/route.ts

```diff
@@ -0,0 +1,107 @@
+import { NextRequest, NextResponse } from "next/server";
+import arcjet, { detectBot, shield, slidingWindow } from "@arcjet/next";
+import { currentUser } from "@clerk/nextjs/server";
+import { Permit } from "permitio";
+
+const aj = arcjet({
+  // Get your site key from https://app.arcjet.com
+  // and set it as an environment variable rather than hard coding.
+  // See: https://nextjs.org/docs/app/building-your-application/configuring/environment-variables
+  key: process.env.ARCJET_KEY!,
+  // Define a global characteristic that we can use to identify users
+  characteristics: ["fingerprint"],
+  rules: [
+    // Shield detects suspicious behavior, such as SQL injection and cross-site
+    // scripting attacks. We want to ru nit on every request
+    shield({
+      mode: "LIVE", // will block requests. Use "DRY_RUN" to log only
+    }),
+  ],
+});
+
+const permit = new Permit({
+  pdp: process.env.PERMIT_PDP!,
+  token: process.env.PERMIT_TOKEN!,
+});
+
+// Returns ad-hoc rules depending on whether the user is logged in, and if they
+// are, whether they have permission to update stats.
+async function getClient() {
+  // If the user is not logged in then give them a low rate limit
+  const user = await currentUser();
+  if (!user) {
+    return aj
+      .withRule(
+        slidingWindow({
+          mode: "LIVE",
+          max: 5,
+          interval: 60,
+        })
+      )
+      .withRule(
+        detectBot({
+          mode: "LIVE",
+          block: ["AUTOMATED"],
+        })
+      );
+  }
+
+  // If the user is logged in but does not have permission to update stats
+  // then give them a medium rate limit.
+  const canUpdate = await permit.check(user.id, "update", "stats");
+  if (!canUpdate) {
+    return aj.withRule(
+      slidingWindow({
+        mode: "LIVE",
+        max: 10,
+        interval: 60,
+      })
+    );
+  }
+
+  // User is logged in and has permission to update stats, so give them no rate limit
+  return aj;
+}
+
+export async function GET(req: NextRequest) {
+  // Get the user's ID if they are logged in, otherwise use their IP address as a fingerprint
+  const user = await currentUser();
+  const fingerprint: any = user ? user.id : req.ip;
```

Comment:
```suggestion
  const fingerprint: string | number = user ? user.id : req.ip;
```
I'm not sure the type of `id` here but `ip` should be string. I'm surprised you need to annotate this at all.